### PR TITLE
Default field values (WIP)

### PR DESCRIPTION
**Pull request by rafiki270**
Adding support for default values for `ColumnType`s

Please let me know if this is something you would consider merging in given it would have all the missing comments and tests.

### CODE REVIEW

**Comment by rafiki270** [comment]
so after discussion with @tanner0101 I will try to add a `defaultValue: Encodable?` to the `public struct SchemaField<Database> where Database: SchemaSupporting` and try to make a foundation for all db's ... or I might create a feature request and ask some proper developer to do this

**Comment by tanner0101** [comment]
Keeping this open until the issue is resolved on Fluent. :)

File: Sources/FluentMySQL/DefaultValue.swift [hunk 1]
```diff
@@ -0,0 +1,31 @@
+import Foundation
+
+
+public enum DefaultValue {
```
Comment (tanner0101) [inline Sources/FluentMySQL/DefaultValue.swift:4]: There should at least be a `raw` case here to pass a custom function. Event better, have `DefaultValue` be a struct and have all of the cases be static conveniences :)

Also, should be `MySQLDefaultValue` for namespacing

```swift
struct MySQLDefaultValue {
    public var value: String

    public static var null: MySQLDefaultValue { return .init(value: "NULL") }
    ... etc
}
```

**Review by tanner0101 — CHANGES_REQUESTED** [review]
We're trying to avoid public enums at all costs. Adding new cases is considered a breaking change.

**Comment by tanner0101** [comment]
This will be possible in 3.0.0 tag by adding custom field types. 👍

**Pull request closed without merging**